### PR TITLE
CS6.2 - Event improvements

### PR DESCRIPTION
**Pull request by kevincheng96**
Some small improvements to events:

- Adds an index on the `wallet` and `nonce` fields in the `NonceSubmitted` event
- Non-replayable operations now emit `EXHAUSTED` as the `submissionToken` value in the `NonceSubmitted` event

### CODE REVIEW

**Review by fluffywaffles — APPROVED** [review]
LGTM, might want to look into the gas snapshot regression. I don't think it's important. The test with a gas use regression is apparently `testClaimsSequentialNonces`, which honestly doesn't even seem like a relevant test anymore based on its name?

File: src/quark-core/src/QuarkNonceManager.sol [hunk 1]
```diff
@@ -71,8 +71,9 @@ contract QuarkNonceManager {
             revert InvalidSubmissionToken(msg.sender, nonce, submissionToken);
         }
 
-        // Note: even with a valid submission token, we always set non-replayables to exhausted (e.g. for cancellations)
-        submissions[msg.sender][nonce] = isReplayable ? submissionToken : EXHAUSTED;
+        // Note: Even with a valid submission token, we always set non-replayables to exhausted (e.g. for cancellations)
+        submissionToken = isReplayable ? submissionToken : EXHAUSTED;
```
Comment (hayesgm) [inline src/quark-core/src/QuarkNonceManager.sol:75]: Can you call this a new variable name?

Comment (kevincheng96) [reply src/quark-core/src/QuarkNonceManager.sol]: Renamed to `finalSubmissionToken`.

File: src/quark-core/src/QuarkNonceManager.sol [hunk 2]
```diff
@@ -19,7 +19,7 @@ contract QuarkNonceManager {
     error InvalidNonce(address wallet, bytes32 nonce);
     error InvalidSubmissionToken(address wallet, bytes32 nonce, bytes32 submissionToken);
 
-    event NonceSubmitted(address wallet, bytes32 nonce, bytes32 submissionToken);
+    event NonceSubmitted(address indexed wallet, bytes32 indexed nonce, bytes32 submissionToken);
```
Comment (kevincheng96) [inline src/quark-core/src/QuarkNonceManager.sol:22]: I also just addded an index on `nonce` since it could be useful for replayable operations.

**Pull request merged**